src/internal/m365/service/groups/backup.go

```diff
@@ -118,5 +129,49 @@ func ProduceBackupCollections(
 		collections = append(collections, baseCols...)
 	}
 
+	// Add metadata about sites
+	md, err := getSitesMetadataCollection(
+		creds.AzureTenantID,
+		bpc.ProtectedResource.ID(),
+		sitesPreviousPaths,
+		su)
+	if err != nil {
+		return nil, nil, false, err
+	}
+
+	collections = append(collections, md)
+
 	return collections, ssmb.ToReader(), canUsePreviousBackup, el.Failure()
 }
+
+func getSitesMetadataCollection(
+	tenantID, groupID string,
+	sites map[string]string,
+	su support.StatusUpdater,
+) (data.BackupCollection, error) {
+	// TODO(meain): Should we store this one level above? If so, how would we
+	// separate different resources? Use different name for each resource?
+	p, err := path.Builder{}.ToServiceCategoryMetadataPath(
+		tenantID,
+		groupID,
+		path.GroupsService,
+		path.LibrariesCategory,
+		false)
+	if err != nil {
+		return nil, clues.Wrap(err, "making metadata path")
+	}
+
+	p, err = p.Append(false, odConsts.SitesPathDir)
+	if err != nil {
+		return nil, clues.Wrap(err, "appending sites to metadata path")
+	}
```

Comment:
Since this pattern is coming up a lot, and feels a bit clunky (imo), thoughts on cleaning it up with a `BuildMetadata` constructor similar to the current [path.Build()](https://github.com/alcionai/corso/blob/main/src/pkg/path/path.go#L134) func?

Reply:
Ah, it came up in my own day.  So here's the change! https://github.com/alcionai/corso/pull/4192

Reply:
Let me add a todo here for now, I'll update it once #4192 is merged.